### PR TITLE
Enhance & Refactor APIs

### PR DESCRIPTION
**Pull request by AbdulazizAlrabiah**

### CODE REVIEW

File: Sdk/MoyasarSdk/Models/CreditCardViewModel.swift [hunk 1]
```diff
@@ -75,9 +75,10 @@ public class CreditCardViewModel: ObservableObject {
     lazy var expiryValidator = ExpiryValidator()
     lazy var securityCodeValidator = SecurityCodeValidator(getNumber: { self.number })
     
-    public init(paymentRequest: PaymentRequest, resultCallback: @escaping ResultCallback) {
+    public init(apiKey: String, paymentRequest: PaymentRequest, resultCallback: @escaping ResultCallback) {
```
Comment (ahoshaiyan) [inline Sdk/MoyasarSdk/Models/CreditCardViewModel.swift:78]: @AbdulazizAlrabiah is there a reason here for taking the `apiKey` separately? Can we instead just add it within the `PaymentRequest` object?

File: Sdk/MoyasarSdk/Models/CreditCardViewModel.swift [hunk 2]
```diff
@@ -75,9 +75,10 @@ public class CreditCardViewModel: ObservableObject {
...
         self.paymentRequest = paymentRequest
         self.resultCallback = resultCallback
+        self.paymentService = PaymentService(apiKey: apiKey)
```
Comment (ahoshaiyan) [inline Sdk/MoyasarSdk/Models/CreditCardViewModel.swift:81]: With the previous change, this line will be as follows:

```swift
self.paymentService = PaymentService(apiKey: paymentRequest.apiKey)
```

File: Sdk/MoyasarSdk/Models/CreditCardViewModel.swift (hunk 1 above)
Comment (ahoshaiyan) [reply Sdk/MoyasarSdk/Models/CreditCardViewModel.swift:78]: Same goes for the `baseUrl`

**Review by ahoshaiyan — CHANGES_REQUESTED** [review]
Excellent updates @AbdulazizAlrabiah, I just have some questions regarding the `apiKey`.

File: Sdk/MoyasarSdk/Models/CreditCardViewModel.swift (hunk 1 above)
Comment (AbdulazizAlrabiah) [reply Sdk/MoyasarSdk/Models/CreditCardViewModel.swift]: Right, it's better that way

Just submitted required changes

**Comment by ahoshaiyan** [comment]
Excellent job @AbdulazizAlrabiah

**Review by ahoshaiyan — APPROVED** [review]

**Comment by ahoshaiyan** [comment]
New version + documentation are public now

**Pull request merged**